tools/schema_registry_tools/schemas_command_generator.py

```diff
@@ -60,21 +59,31 @@ def generate_options():
     j = json.loads(tmp)
 
     # Generating rpk topic create commands
+    expected_offset = 0
     cmd_all = ''
     cmd_all += f"#!/bin/bash\n\n"
     for c, i in enumerate(j):
         d = json.loads(i['key'])
-        if d['seq'] > i['offset']:
-            cmd_all = f"The seq {d['seq']} is unexpectedly higher than the \
-offset {i['offset']} at key {i['key']}. \nThat is it's likely broken, hence exiting...."
+        if i['offset'] > expected_offset:
+            gap = i['offset'] - expected_offset
+            for g in range(gap):
+                cmd_all += 'echo \'\' | rpk topic produce _schemas --compression none -k \'{"keytype":"NOOP","magic":0}\'\n'
+                cmd_all += "sleep 1s\n"
+                expected_offset += 1
+        elif i['offset'] < expected_offset:
+            cmd_all = f"The offset number doesn't increase monotonically at key {i['key']}.\n\
+This is unexpected and we're exiting without generating the commands."
```

Comment:
I just added this part as a safe-guard but this shouldn't happen unless Redpanda has a bug. I'd like to see how @BenPope thinks.

Reply:
It would imply that the file is in the wrong order. It could be sorted by offset to avoid that.

This tool could also perform the backup.

Reply:
My intention was the offset sequence looks like [0, 1, 2, 3, 2, 3, 4]. In this case, it's obviously broken, right? Your implication would be [0, 1, 2, 3, 5, 4, 6], correct? In this case I'd understand offset 5 and 4 should be swapped.

Reply:
Yeah, essentially there could be a validation/clean phase. It should sort by offset and check they're unique.

Reply:
Sounds good. Will sort it out.

Reply:
Fixed in eb7447b.